lib/rules/no-restricted-class.js

```diff
@@ -0,0 +1,221 @@
+ * @param {BinaryExpression | Expression} left
+ * @param {BinaryExpression | Expression} right
+ * @returns {string}
+ */
+const flattenBinaryExpression = (left, right) => {
+  const result = []
+  if (left) {
+    if (left.type === 'Literal') {
+      result.push(left.value)
+    } else if (
+      left.type === 'BinaryExpression' &&
+      left.left.type !== 'PrivateIdentifier'
+    ) {
+      result.push(flattenBinaryExpression(left.left, left.right))
+    }
+  }
+
+  if (right) {
+    if (right.type === 'Literal') {
+      result.push(right.value)
+    } else if (
+      right.type === 'BinaryExpression' &&
+      right.left.type !== 'PrivateIdentifier'
+    ) {
+      result.push(flattenBinaryExpression(right.left, right.right))
+    }
+  }
+
+  return result.join(' ')
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow specific classes in Vue components',
+      url: 'https://eslint.vuejs.org/rules/no-restricted-class.html',
+      categories: undefined
+    },
+    fixable: null,
+    messages: {
+      forbiddenClass: "'{{class}}' class is not allowed."
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          classes: {
+            type: 'array',
+            items: {
+              type: 'string'
+            }
+          },
+          files: {
+            type: 'array',
+            items: { type: 'string' }
+          }
+        },
+        additionalProperties: false
+      }
+    ]
+  },
+
+  /** @param {RuleContext} context */
+  create(context) {
+    const config = context.options[0] || {}
+    const forbiddenClasses = new Set(config.classes || [])
+    const forbiddenClassFiles = config.files || []
+
+    forbiddenClassFiles.forEach((/** @type {string} */ filePath) => {
+      if (!fs.existsSync(filePath)) {
+        throw new Error(`File "${filePath}" does not exist.`)
+      }
+      const rawData = fs.readFileSync(filePath, 'utf8')
+      JSON.parse(rawData).forEach((/** @type {string} */ forbiddenClass) => {
+        forbiddenClasses.add(forbiddenClass)
+      })
+    })
+
+    return utils.defineTemplateBodyVisitor(context, {
+      // <Element class="forbidden" />
+      /**
+       * @param {VAttribute & { value: VLiteral } } node
+       */
+      'VAttribute[directive=false][key.name="class"]'(node) {
+        const classes = new Set(node.value.value.split(/\s+/))
+        reportForbiddenClasses(classes, node, context, forbiddenClasses)
+      },
+
+      // <Element :class="{forbidden: someBoolean}" />
+      /**
+       * @param {VAttribute & { value: { expression: ObjectExpression } } } node
+       */
+      'VAttribute[directive=true][key.argument.name="class"][value.expression.type="ObjectExpression"]'(
```

Comment:
`[key.name.name='bind']` is required. If not, `@class="..."` will also be hit.
```suggestion
      'VAttribute[directive=true][key.name.name="bind"][key.argument.name="class"][value.expression.type="ObjectExpression"]'(
```